Enforce dependencies check at runtime

mougams opened the pull request:
Currently the binaries version are only checked when called
with flag "version". Now enforce it so that it fails with explicit
error message in case of missing binary or incorrect version installed.

pierrecdn reviewed (commented):
The comment currently doesn't help me to understand what this `fixMajorVersion` flag really is.
Could you clarify?

mougams commented:
> The comment currently doesn't help me to understand what this `fixMajorVersion` flag really is. Could you clarify?

I've simplified the patch - now only enforcing the version check at runtime. Removed the other part.

pierrecdn commented on haproxy/haproxy_cmd/run.go:
```diff
@@ -128,17 +128,20 @@ func CheckEnvironment(dataplaneapiBin, haproxyBin string) error {
 		currVer, e := getVersion(path)
 		if e != nil {
 			err = e
+			return
```
* `return nil` ? (to explicitly match the contract, as it done at the end)
* do we need to "break" the flow here instead of doing what was done before?

mougams replied:
> * `return nil` ? (to explicitly match the contract, as it done at the end)
The `return nil` was for the parent function - here we want to exit the nest function.
Also, the nested function doesn't return anything, it just initializes the err variable from the parent function.

>     * do we need to "break" the flow here instead of doing what was done before?
Yes, because, we want to print the error in case of error other than the version comparison. For ex, when the binary doesn't exist.

pierrecdn replied:
* true, my bad.
* Ack!
